grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy

```diff
@@ -684,22 +688,22 @@ class ProductAvailabilityService {
         return data
     }
 
-    Map<InventoryItem, Integer> getQuantityOnHandByInventoryItem(Location location) {
-        def quantityMap = [:]
-        if (location) {
-            def results = ProductAvailability.executeQuery("""
-						select ii, sum(pa.quantityOnHand)
-						from ProductAvailability pa, InventoryItem ii
-						where pa.location = :location
-						and pa.inventoryItem = ii
-						group by ii
-						""", [location: location])
-
-            results.each {
-                quantityMap[it[0]] = it[1]
+    Map<InventoryItem, Integer> getQuantityOnHandByInventoryItem(Location location, List<InventoryItem> inventoryItems = []) {
+        if (!location) {
+            return []
```

Comment:
this method by default was returning an empty Map, not a List - even the returned type suggests it should be the Map.
I think this would not work as expected with the list.

Reply:
thanks, there was a similar case that returns a list so I messed it up.